File: AdkTabber/src/interfaces/song.ts
import ITab from "./tab";

export default interface ISong {
    id: number;
    name: string;    
    band: string;
    tempo: number;
    tabs: ITab[];
    difficulty : DifficultyType;
}

export enum DifficultyType{
    Easy,
    Medium,
    Hard,
    Any
}

File: Model/SongModel/Song.cs
using Model.TabModel;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Model.SongModel
{
    public class Song
    {
        [Key]
        public int ID { get; set; }

        public string Name { get; set; }

        public string Band { get; set; }

        public int Tempo { get; set; }

        public DifficultyType Difficulty { get; set; }

        public List<Tab> Tabs { get; set; }
    }
}
